### PR TITLE
[Docs] Add story: shopping cart

### PR DESCRIPTION
**Pull request by pvcarrera**
This PR copies shopping cart exercise over from the research repository. I think this exercise is complex enough to teach a bunch of things 😄 But probably the main is dealing with the state of the cart

### CODE REVIEW

**Comment by pvcarrera** [comment]
I run `./bin/format.sh` but prettier is still complaining, I'm not sure what I'm missing here 😄

File: reference/stories/shopping-cart.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Shopping cart exercise
+
+There are multiple ways of solving this exercise. It's probably necessary to master some basic concepts like: strings, hashes or collections before this exercise can be tackle.
```
Comment (kotp) [inline reference/stories/shopping-cart.md:3]:
```suggestion
There are multiple ways of solving this exercise. It's probably necessary to master some basic concepts like: strings, hashes or collections before this exercise can be tackled.
```

Comment (pvcarrera) [reply reference/stories/shopping-cart.md]: Thanks @kotp !

**Comment by wolf99** [comment]
Looks good!

File: reference/stories/shopping-cart.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+| Coffee    | 14.99 | STACOF |
+| Newspaper | 2.99  | MEDNEW |
+
+You need to define a `ShoppingCart` class, with three public methods:
+
+- `add`: Takes a SKU and adds it to the cart. The SKU is a `Symbol` comprised of the first three uppercase letters of the product type + the first three uppercase letters of the product name.
+- `total_amount`: The value of the items in the cart as a number with two decimal places (e.g. 3.99).
+- `items_list`: A list of the names of the items in the cart, ordered alphabetically, with no duplicates, separated by a comma and a space.
+
+For example:
```
Comment (SleeplessByte) [inline reference/stories/shopping-cart.md:25]: I'd leave the ruby code out, but instead link to the ruby example implementation / exercise, like [this](https://github.com/exercism/v3/blob/master/reference/stories/arrays.magician-card-stack.md#tasks)

**Comment by ErikSchierboom** [comment]
@pvcarrera Thanks for doing this! I've made some consistency changes to the story format this morning, which can be found [in this PR](https://github.com/exercism/v3/pull/1408). Note that the only section that is required is the `## Story` section. The other sections are all optional.

**Comment by pvcarrera** [comment]
@ErikSchierboom Thanks for keeping the repos consistent 🙇‍♂️ I adjusted the format and added the exercise extension. I'm still not sure which concept/concepts will go together with this exercise, for now, I added `state` as the concept on the stories menu.

**Pull request merged**